recipes/llvm-core/all/conanfile.py

```diff
@@ -1,429 +1,510 @@
+from conan.errors import ConanInvalidConfiguration
+from conan.tools.apple import is_apple_os
+from conan.tools.build import check_min_cppstd, can_run, cross_building
+from conan.tools.cmake import CMake, CMakeDeps, CMakeToolchain, cmake_layout
+from conan.tools.env import VirtualRunEnv
+from conan.tools.files import (
+    apply_conandata_patches,
+    collect_libs,
+    get,
+    rmdir,
+    save,
+    copy,
+    export_conandata_patches,
+    rm,
+    rename,
+    replace_in_file
+)
+from conan.tools.microsoft import is_msvc
 from conan.tools.scm import Version
-from conans import CMake
-from collections import defaultdict
+
 import json
-import re
-import os.path
 import os
+from pathlib import Path
+import re
 import textwrap
 
-required_conan_version = ">=1.50.2" # Due to conan.tools.scm.Version
+
+required_conan_version = ">=1.62.0"
+
+# LLVM's default config is enable all targets, but end users can significantly reduce
+# build times for the package by disabling the ones they don't need with the corresponding option
+# `-o llvm-core/*:with_target_<target name in lower case>=False`
+LLVM_TARGETS = [
+    "AArch64",
+    "AMDGPU",
+    "ARM",
+    "AVR",
+    "BPF",
+    "Hexagon",
+    "Lanai",
+    "LoongArch",
+    "Mips",
+    "MSP430",
+    "NVPTX",
+    "PowerPC",
+    "RISCV",
+    "Sparc",
+    "SystemZ",
+    "VE",
+    "WebAssembly",
+    "X86",
+    "XCore"
+]
 
 
 class LLVMCoreConan(ConanFile):
-    name = 'llvm-core'
+    name = "llvm-core"
     description = (
-        'A toolkit for the construction of highly optimized compilers,'
-        'optimizers, and runtime environments.'
+        "A toolkit for the construction of highly optimized compilers,"
+        "optimizers, and runtime environments."
     )
-    license = 'Apache-2.0 WITH LLVM-exception'
-    topics = ('llvm', 'compiler')
-    homepage = 'https://llvm.org'
-    url = 'https://github.com/conan-io/conan-center-index'
-
-    settings = ('os', 'arch', 'compiler', 'build_type')
+    license = "LLVM-exception"
+    topics = ("llvm", "compiler")
+    homepage = "https://llvm.org"
+    url = "https://github.com/conan-io/conan-center-index"
+    settings = "os", "arch", "compiler", "build_type"
     options = {
-        'shared': [True, False],
-        'fPIC': [True, False],
-        'components': 'ANY',
-        'targets': 'ANY',
-        'exceptions': [True, False],
-        'rtti': [True, False],
-        'threads': [True, False],
-        'lto': ['On', 'Off', 'Full', 'Thin'],
-        'static_stdlib': [True, False],
-        'unwind_tables': [True, False],
-        'expensive_checks': [True, False],
-        'use_perf': [True, False],
-        'use_sanitizer': [
-            'Address',
-            'Memory',
-            'MemoryWithOrigins',
-            'Undefined',
-            'Thread',
-            'DataFlow',
-            'Address;Undefined',
-            'None'
+        "shared": [True, False],
```

Comment:
This pull request is a night mare to review due to size and patches, you really shouldn't have made formatting changes in such a pull request 🙈

Reply:
Sorry that it's been a "nightmare" for you, but the recipe was unmaintained and did not build with conan2. Now it does.

Reply:
Thank you for your work! Didn't mean to sound ungrateful or something 🙂